src/main/java/frc/robot/subsystems/swerve/SwerveModule.java

```diff
@@ -147,4 +147,56 @@ public static Pair<Double, Double> optimizeModuleState(SwerveModuleState target,
     private Rotation2d getAngle() {
         return new Rotation2d(steerEncoder.getPosition() + offsetRads);
     }
+
+    /**
+     * Utility class to construct a top left swerve module from a module's pin offset.
+     * The offset to align the pin with the front of the robot is automatically applied.
+     */
+    public static class TopLeft extends SwerveModule {
+        public TopLeft(int drivePort, int steerPort) {
+            super(drivePort, steerPort, 0.0);
+        }
+        public TopLeft(int drivePort, int steerPort, double offsetRads) {
+            super(drivePort, steerPort, offsetRads);
+        }
+    }
+
+    /**
+     * Utility class to construct a top right swerve module from a module's pin offset.
+     * The offset to align the pin with the front of the robot is automatically applied.
+     */
+    public static class TopRight extends SwerveModule {
```

Comment:
would it be cleaner to use an enum in SwerveModule's constructor instead of having four subclasses?

Reply:
Perhaps, but this way we can typecheck the alignment of the modules in `SwerveSubsystem` so that it generates a compile time error of someone accidentally constructs a top right module for the top left module variable.

Reply:
Maybe if that's not necessary an enum would too.

Reply:
Hmm, I guess that's nice. Either way it doesn't matter too much, it's compile time errors vs a couple less class declarations, so it's honestly up to you